ui/apps/platform/src/Containers/NetworkGraph/NetworkGraphPage.tsx

```diff
@@ -386,60 +364,21 @@ function NetworkGraphPage() {
                     <Divider component="div" />
                 </>
             )}
-            <PageSection
-                className="network-graph"
-                variant={hasClusterNamespaceSelected ? 'default' : 'light'}
-                padding={{ default: 'noPadding' }}
-            >
-                {hasReadAccessForGenerator && !hasClusterNamespaceSelected && (
```

Comment:
Hmm this is definitely an inconsistency - the `hasReadAccess('Integration')` is needed for the "Share with Notifier" action within the side panel, and isn't referenced in the `TopologyComponent` rendering of the side panel.

There is a check for the `Integration` permission before displaying the "Network Policy Generator" button at the top of the page, which is currently the only way to open that panel. It is safe for now but a change in behavior could make this fall out of sync.

What do you think of a follow up to move the `hasReadAccess('Integration')` check into the sidebar component itself and only use it to conditionally render the Notifier menu option? It seems that would be safer than relying on the top level check here and would give more functionality to users without that permission, while still restricting the UI.

Reply:
https://issues.redhat.com/browse/ROX-21673

Reply:
Super, thank you for confirming that top level check was temporary (but no comment, my bad) until we can move conditionally rendering to lowest level that prevents interaction which would cause request that has 403 Forbidden status.